zerver/management/commands/enqueue_digest_emails.py

```diff
@@ -22,5 +25,13 @@ class Command(BaseCommand):
 
     @override
     def handle(self, *args: Any, **options: Any) -> None:
-        cutoff = timezone_now() - timedelta(days=DIGEST_CUTOFF)
-        enqueue_emails(cutoff)
+        with lockfile_nonblocking(settings.ENQUEUE_DIGEST_EMAILS_LOCK_FILE) as lock_acquired:
+            if not lock_acquired:
+                print(
+                    f"{WARNING}Lock {settings.ENQUEUE_DIGEST_EMAILS_LOCK_FILE} is unavailable;"
+                    f" exiting.{ENDC}"
+                )
+                sys.exit(0)
```

Comment:
I like the early return better than the deeper nesting.

That said, this pattern might be extractable into a decorator for the `handle` method, that accepts the lock file as a parameter? I'm thinking that might feel nicer as a pattern.

Reply:
Yeah, that's part of the cleanup.  Decorator is a good idea -- it can't be a contextmanager, because those must yield to the body, but decorator should work.